plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java

```diff
@@ -954,6 +959,25 @@ public ListObjectsV2Result listObjectsV2(ListObjectsV2Request listObjectsV2Reque
         }
     }
 
+    @Test
+    public void testThatTrinoS3FileSystemReportsConsumedMemory()
+            throws IOException
+    {
+        TestMemoryReservationHandler memoryReservationHandler = new TestMemoryReservationHandler();
+        AggregatedMemoryContext memoryContext = newRootAggregatedMemoryContext(memoryReservationHandler, 1024 * 1000 * 1000);
+        try (TrinoS3FileSystem fs = new TrinoS3FileSystem()) {
+            MockAmazonS3 s3 = new MockAmazonS3();
+            Path rootPath = new Path("s3n://test-bucket/");
+            fs.initialize(rootPath.toUri(), newEmptyConfiguration());
+            fs.setS3Client(s3);
+            OutputStream outputStream = fs.create(new Path("s3n://test-bucket/test1"), memoryContext);
+            outputStream.write(new byte[] {1, 2, 3, 4, 5, 6}, 0, 6);
+            outputStream.close();
+        }
+        assertThat(memoryReservationHandler.getReserved()).isEqualTo(0);
+        assertThat(memoryReservationHandler.getSetValues()).contains(64L);
```

Comment:
I would just store `max` and make sure that `assertThat(memoryReservationHandler.getMaxReserver).isGreatherThan(0);`

Checking that 64 is part of reserved values is fragile (depends on implementation)